presto-hive/src/main/java/com/facebook/presto/hive/HiveColumnHandle.java

```diff
@@ -33,6 +35,15 @@
         implements ColumnHandle
 {
     public static final String SAMPLE_WEIGHT_COLUMN_NAME = "__presto__sample_weight__";
+
+    // columns with column index < -10 will be hidden
+    public static final int HIDDEN_COLUMN_INDEX_THRESHOLD = -10;
```

Comment:
This is unnecessary. `$shard_row_id` is a hidden column. It's special in that it only has partial implementation because full implementation was not necessary.